src/store/ChartState.js

```diff
@@ -393,29 +395,14 @@ class ChartState {
                 );
                 this.stxx.createDataSet();
             }
-            this.stxx.maxMasterDataSize = 0;
-            const scrollAnimator = new CIQ.EaseMachine(Math.easeOutCubic, 1000);
-            const scrollToTarget = this.stxx.chart.dataSegment.length;
-            scrollAnimator.run((bar) => {
-                bar = Math.ceil(bar); // round-up for precision
-                const scroll = this.stxx.chart.dataSegment.length - bar;
-                if (scroll <= 2 || bar === scrollToTarget) {
-                    /**
-                     * Stop scrolling and draw markers if
-                     * the scroll value is off-screen or if the animator has reached target.
-                     * We check scroll <= '2' because sometimes the chart is scrolled so that the first
-                     * bar is partially hidden off-screen
-                     */
-                    scrollAnimator.stop();
-                    this.stxx.chart.entryTick = this.stxx.tickFromDate(startEntry.DT);
-                    this.stxx.chart.lockScroll = true;
-                    this.stxx.chart.isScrollLocationChanged = true; // set to true to draw markers
-                } else {
-                    this.stxx.chart.scroll = scroll;
-                }
-            },
-            0, scrollToTarget);
-            this.stxx.draw();
+
+            const tick = this.stxx.tickFromDate(startEntry.DT);
+            const startTick = this.stxx.chart.dataSet.length - tick;
+            this.stxx.chart.lockScroll = true;
+            this.stxx.chart.isScrollLocationChanged = true; // set to true to draw markers
+
+            if (!this.startEpoch) this.scrollChartToLeftAnimator(startTick, startEntry);
+            else this.scrollChartToLeftGraph(startTick, startEntry);
```

Comment:
@shayanbinary is this `else` clause necessary?  shouldn't the `scrollChartToLeftAnimator` takes care of it? i think charts need to wait until it gets `startEpoch` value to start the scrolling animation